src/cpp/src/visual_language/vision_encoder.cpp

```diff
@@ -417,23 +410,81 @@ ov::Tensor preprocess_image_llava(const ov::Tensor& image, const ProcessorConfig
     std::copy(config.image_std.begin(), config.image_std.end(), ctx.image_std);
 
     clip_image_f32 normalized_image = clip_image_preprocess(ctx, cropped_image);
+    return normalized_image;
+}
+
+ov::Tensor get_pixel_values_llava(const ov::Tensor& image, const ProcessorConfig& config) {
+    // ov::Tensor to clip_image_u8
+    clip_image_u8 input_image{
+        int(image.get_shape().at(3)),
+        int(image.get_shape().at(2)),
+        {image.data<uint8_t>(), image.data<uint8_t>() + image.get_size()}
+    };
+
+    clip_image_f32 preprocessed_image = preprocess_clip_image_llava(input_image, config);
 
     // Convert clip_image_f32 to ov::Tensor
     ov::Tensor result(
         ov::element::f32,
-        {1, 3, size_t(normalized_image.ny), size_t(normalized_image.nx)},
-        (void*)(normalized_image.buf.data())
+        {1, 3, size_t(preprocessed_image.ny), size_t(preprocessed_image.nx)},
+        (void*)(preprocessed_image.buf.data())
     );
 
     return result;
 }
+
+ov::Tensor get_pixel_values_llava_next(const ov::Tensor& image, const ProcessorConfig& config) {
+    // TODO Consider moving to separate function and reuse
+    // ov::Tensor to clip_image_u8
+    clip_image_u8 input_image{
+        int(image.get_shape().at(3)),
+        int(image.get_shape().at(2)),
+        {image.data<uint8_t>(), image.data<uint8_t>() + image.get_size()}
+    };
+
+    std::pair<int, int> size{config.size_shortest_edge, config.size_shortest_edge};
+    auto patch_size = config.crop_size_height;
+    auto image_patches = get_image_patches(input_image, config.image_grid_pinpoints, size, patch_size);
+
+    // Preprocess image patches
+    std::vector<clip_image_f32> processed_patches;
+    processed_patches.reserve(image_patches.size());
+
+    for (const auto& patch : image_patches) {
+        processed_patches.push_back(preprocess_clip_image_llava(patch, config));
```

Comment:
is it possible to process image patches as a single batch ? just as a future optimization

Reply:
Added to PR TODOS